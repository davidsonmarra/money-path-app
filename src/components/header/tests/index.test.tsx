import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Header, {Props as HeaderProps} from 'src/components/header';

jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: () => ({top: 0}),
}));

const mockText = 'Snapshot test!';
const renderComponent = ({text = mockText, ...rest}: HeaderProps) =>
  render(<Header text={text} {...rest} />);

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    renderComponent({});
    const headerText = screen.getByText(mockText);
    expect(headerText.children).toEqual([mockText]);
  });

  it('should call onLeftIconPress', () => {
    const onPress = jest.fn();
    const {getByTestId} = renderComponent({onLeftIconPress: onPress});
    const leftIcon = getByTestId('btn-left-icon');
    leftIcon.props.onClick();
    expect(onPress).toHaveBeenCalled();
  });
});
