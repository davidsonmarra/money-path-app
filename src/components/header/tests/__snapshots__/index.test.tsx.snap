// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2A5C99",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="btn-left-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#FFF"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={960}
      vbWidth={960}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      [
        {
          "color": undefined,
          "fontSize": 20,
          "fontWeight": "bold",
        },
        {
          "color": undefined,
        },
      ]
    }
  >
    Snapshot test!
  </Text>
  <View />
</View>
`;
